### PR TITLE
Allow Tox Environments to be Configured As Skippable via platform-matrix

### PR DESCRIPTION
**Pull request by scbedd**
This PR will allow us to resolve the nightly failure that identity is currently hitting every day.

It also:

- updates identity matrix to exclude `mindependency/latestdependency` checks for the MSAL run
- updates an inline pshell script to a python script

### CODE REVIEW

File: eng/pipelines/templates/jobs/ci.tests.yml [hunk 1]
```diff
@@ -79,20 +82,14 @@ jobs:
...
-
-        # ensure that the variable is unset. if it isn't, use the value discovered there
-        if ('$(Run.ToxCustomEnvs)' -ne ('$' + '(Run.ToxCustomEnvs)'))
-        {
-          $toxenvvar = '$(Run.ToxCustomEnvs)'
-        }
-
-        echo "##vso[task.setvariable variable=toxenv]$toxenvvar"
-      displayName: "Set Tox Environment"
+    - task: PythonScript@0
```
Comment (mccoyp) [inline eng/pipelines/templates/jobs/ci.tests.yml:85]: Out of curiosity, why the "@0"?

**Review by mccoyp — APPROVED** [review]

File: eng/pipelines/templates/jobs/ci.tests.yml (hunk 1 above)
Comment (semick-dev) [reply eng/pipelines/templates/jobs/ci.tests.yml:85]: It's the version of the job!  The way that you access native devops tasks is by the name + the version. [here is the index of native tasks](https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks)

image

**Review by chlowell — APPROVED** [review]
👍

**Pull request merged**